src/server/system_services/pool_server.js

```diff
@@ -652,7 +652,9 @@ async function update_hosts_pool(req) {
 
 function delete_pool(req) {
     const pool = find_pool_by_name(req);
-    check_deletion_ownership(req, pool.owner_id);
+    // rebuild_object_links() resolves the pool's owner_id to the account object
+    // which is why we have to access ._id to get the actual ID
+    check_deletion_ownership(req, pool.owner_id._id);
```

Comment:
do we always have `owner_id`? 
maybe better to check 

```suggestion
    check_deletion_ownership(req, pool.owner_id?._id);
```